pkg/controller/resource/resource.go

```diff
@@ -26,10 +27,25 @@ import (
 
 // Annotations on app to enable chaos on it
 const (
-	ChaosAnnotationKey   = "litmuschaos.io/chaos"
 	ChaosAnnotationValue = "true"
 )
 
+var (
+	ChaosAnnotationKey = getCustomAnnotation()
```

Comment:
This should be called only when ChaosAnnotaionValue, will be true ?

Reply:
ChaosAnnotationKey is just the lvalue, and ChaosAnnotationValue (rvalue) only supports `true`

Reply:
getCustomAnnotation() returns the annotation from the ENV if set, else use the default one.

Reply:
I guess @sumitnagal is asking that this check/key will come into play only if the `.spec.annotationCheck` in the chaosEngine is set to `true`. If that is the question, yes that is the case.

Reply:
In this PR @rahulchheda provides for a custom annotation key, while expecting the value is set to `true` (as a const) -- as in `litmuschaos.io/chaos: true"